packages/expo-router/src/getDevServer/index.ts

```diff
@@ -21,15 +20,11 @@ export const getDevServer = () => {
         return document.currentScript.src;
       }
 
-      const url = window.location.toString();
-      const query = qs.parse(url);
+      const query = new URLSearchParams(location.search);
 
-      return (
-        location.origin +
-        location.pathname +
-        '?' +
-        qs.stringify({ ...query, platform: Platform.OS })
-      );
+      query.append('platform', Platform.OS);
+
+      return location.origin + location.pathname + '?' + query;
```

Comment:
Would it be better to preserve the whole URL 1:1 (including the hash fragment) and just add a query parameter?

```ts
const url = new URL(location);
url.searchParams.append('platform', Platform.OS);  // or "set" perhaps?
return url.toString();
```

Reply:
That could lead to a lot of side-effects, this URL is just used for opening web socket connections and other dev-tool related features.

Reply:
What side effects would there be? The URL instance would still be a new one and not modify the location object. The main difference is that the hash fragment would get carried over and instead of needing to write string concatenations, the URL library would take care of making the final href string.